src/benchmark.cc

```diff
@@ -313,15 +310,13 @@ class BenchmarkImp {
   friend class BenchmarkFamilies;
 
   std::string name_;
-  Function* function_;
   int arg_count_;
   std::vector< std::pair<int, int> > args_;  // Args for all benchmark runs
   double min_time_;
   bool use_real_time_;
   std::vector<int> thread_counts_;
-  std::size_t registration_index_;
 
-  BENCHMARK_DISALLOW_COPY_AND_ASSIGN(BenchmarkImp);
+  BenchmarkImp& operator=(BenchmarkImp const&);
```

Comment:
Unfortunately that would require

1. Passing std::unique_ptr across the library boundary.
2. C++11 in the headers.